www/codePush.ts

```diff
@@ -34,26 +34,86 @@ class CodePush implements CodePushCordovaPlugin {
      */
     private static DefaultUpdateDialogOptions: UpdateDialogOptions;
     
-    /**
-      * Notifies the plugin that the update operation succeeded and that the application is ready.
-      * Calling this function is required if a rollbackTimeout parameter is used for your LocalPackage.install() call.
-      * If install() is used without a rollbackTimeout, calling this function is a noop.
-      * 
-      * @param notifySucceeded Optional callback invoked if the plugin was successfully notified.
-      * @param notifyFailed Optional callback invoked in case of an error during notifying the plugin.
-      */
+    /**  
+     * Notifies the plugin that the update operation succeeded and that the application is ready.
+     * Calling this function is required on the first run after an update. On every subsequent application run, calling this function is a noop.
+     * If using sync API, calling this function is not required since sync calls it internally. 
+     * 
+     * @param notifySucceeded Optional callback invoked if the plugin was successfully notified.
+     * @param notifyFailed Optional callback invoked in case of an error during notifying the plugin.
+     */
     public notifyApplicationReady(notifySucceeded?: SuccessCallback<void>, notifyFailed?: ErrorCallback): void {
         cordova.exec(notifySucceeded, notifyFailed, "CodePush", "updateSuccess", []);
     }
     
     /**
-    * Get the current package information.
-    * 
-    * @param packageSuccess Callback invoked with the currently deployed package information.
-    * @param packageError Optional callback invoked in case of an error.
-    */
+     * Reloads the application. If there is a pending update package installed using ON_NEXT_RESTART or ON_NEXT_RESUME modes, the update
+     * will be immediately visible to the user. Otherwise, calling this function will simply reload the current version of the application.
+     */
+    public restartApplication(installSuccess: SuccessCallback<void>, errorCallback?: ErrorCallback): void {
+        cordova.exec(installSuccess, errorCallback, "CodePush", "restartApplication", []);
+    }
+    
+    /**
+     * Reports an application status back to the server.
+     * !!! This function is called from the native side, please make changes accordingly. !!!
+     */
+    public reportStatus(status: number, label: string, appVersion: string) {
```

Comment:
I just pushed a change that handles this. It also handles the download reporting. Please take a look.